pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -15,24 +15,100 @@
 Contains the MPSPrep template.
 """
 
+import numpy as np
+
 import pennylane as qml
 from pennylane.operation import Operation
 from pennylane.wires import Wires
 
 
+def right_canonicalize_mps(mps):
+    """Transform an MPS into a right-canonical MPS.
+
+    Args:
+        mps (list[TensorLike]): List of tensors representing the MPS.
+
+    Returns:
+        A list of tensors representing the MPS in right-canonical form.
+
+    .. seealso:: :class:`~.MPSPrep`.
+
+    """
+
+    is_right_canonical = True
+    d_shapes = []
+    for tensor in mps[1:-1]:
+        # Right-canonical definition
+        d_shapes += tensor.shape
+        M = qml.math.tensordot(tensor, tensor.conj(), axes=([1, 2], [1, 2]))
+        if not qml.math.allclose(M, qml.math.eye(tensor.shape[0])):
+            is_right_canonical = False
+
+    if is_right_canonical:
+        return mps
+
+    max_bond_dim = qml.math.max(d_shapes)
+
+    L = len(mps)
+    output_mps = [None] * L
+
+    # Procedure analogous to the left-canonical conversion but starting from the right and storing the Vd
+    for i in range(L - 1, 0, -1):
+        chi_left, d, chi_right = mps[i].shape
+        M = mps[i].reshape(chi_left, d * chi_right)
+
+        # Perform Singular Value Decomposition (SVD)
+        # M ≈ U @ diag(S) @ Vd, where:
+        # - U represents an orthogonal rotation in the space of chi_left
+        # - diag(S) scales the singular values (stretching or contracting)
+        # - Vd represents another rotation in the space of d * chi_right
```

Comment:
Is this informative? Reader can see that `qml.math.linalg.svd` performs singular value decomposition.